### ISSUE TITLE
When parsing an Object, keys with undefined value should only be returned if they are provided in input

### ISSUE THREAD

**Issue by xinha-sh** [fabian-hiller/valibot#199]
### Discussed in https://github.com/fabian-hiller/valibot/discussions/196

<div type='discussions-op-text'>

<sup>Originally posted by **xinha-sh** October  4, 2023</sup>
```ts
import * as v from "valibot";

const baseSchema = v.object({
  x: v.optional(v.array(v.string())),
  y: v.optional(v.array(v.string())),
 z: v.optional(v.array(v.string())),
})

const schemaA = v.object({
  key1: v.optional(baseSchema), 
  key2: v.optional(baseSchema)
})

const input = {key1:  {x: ["hello", "world"]}}
const result = v.parse(schemaA, input)

// result.output
// {key1: {x:  ["hello", "world"]}, key2: undefined}
```

Why do I get ``key2: undefined`` in the output ? What's the best way to ignore them.

My use case is that upon successful parsing I want to loop over array values via ``Object.values(catalog).map(item => objectKeys(item)).flat()`` which breaks with error ``Cannot convert undefined or null to object``

This used to work as intended with zod schema. 
Unless my input is ``const input = {key1:  {x: ["hello", "world"]}, key2: undefined}`` the output shouldn't contain undefined.

What's valibot's way of stripping away undefined keys ?</div>

**Comment by fabian-hiller** [comment]
Thank you! I will investigate this.

**Comment by fabian-hiller** [comment]
Fixed in v0.19.0
